code/include/playfab/PlayFabProfilesDataModels_c.h

```diff
@@ -0,0 +1,259 @@
+
+typedef struct PlayFabProfilesGetEntityProfileResponse
+{
+    PF_OPTIONAL PlayFabProfilesEntityProfileBody* profile;
+
+} PlayFabProfilesGetEntityProfileResponse;
+
+typedef struct PlayFabProfilesGetEntityProfilesRequest
+{
+    PF_MAP struct PlayFabStringDictionaryEntry* customTags;
+    PF_COLLECTION_COUNT uint32_t customTagsCount;
+    PF_OPTIONAL bool* dataAsObject;
+    PF_ARRAY PlayFabProfilesEntityKey** entities;
+    PF_COLLECTION_COUNT uint32_t entitiesCount;
+
+} PlayFabProfilesGetEntityProfilesRequest;
+
+typedef struct PlayFabProfilesGetEntityProfilesResponse
+{
+    PF_ARRAY PlayFabProfilesEntityProfileBody** profiles;
+    PF_COLLECTION_COUNT uint32_t profilesCount;
+
+} PlayFabProfilesGetEntityProfilesResponse;
+
+typedef struct PlayFabProfilesGetGlobalPolicyRequest
+{
+    PF_MAP struct PlayFabStringDictionaryEntry* customTags;
+    PF_COLLECTION_COUNT uint32_t customTagsCount;
+
+} PlayFabProfilesGetGlobalPolicyRequest;
+
+typedef struct PlayFabProfilesGetGlobalPolicyResponse
+{
+    PF_ARRAY PlayFabProfilesEntityPermissionStatement** permissions;
+    PF_COLLECTION_COUNT uint32_t permissionsCount;
+
+} PlayFabProfilesGetGlobalPolicyResponse;
+
+typedef struct PlayFabProfilesGetTitlePlayersFromMasterPlayerAccountIdsRequest
+{
+    PF_MAP struct PlayFabStringDictionaryEntry* customTags;
+    PF_COLLECTION_COUNT uint32_t customTagsCount;
+    PF_ARRAY const char** masterPlayerAccountIds;
+    PF_COLLECTION_COUNT uint32_t masterPlayerAccountIdsCount;
+    const char* titleId;
+
+} PlayFabProfilesGetTitlePlayersFromMasterPlayerAccountIdsRequest;
+
+typedef struct PlayFabProfilesGetTitlePlayersFromMasterPlayerAccountIdsResponse
+{
+    const char* titleId;
+    PF_MAP struct PlayFabProfilesEntityKeyDictionaryEntry* titlePlayerAccounts;
+    PF_COLLECTION_COUNT uint32_t titlePlayerAccountsCount;
+
+} PlayFabProfilesGetTitlePlayersFromMasterPlayerAccountIdsResponse;
+
+typedef struct PlayFabProfilesSetEntityProfilePolicyRequest
+{
+    PF_MAP struct PlayFabStringDictionaryEntry* customTags;
+    PF_COLLECTION_COUNT uint32_t customTagsCount;
+    PlayFabProfilesEntityKey* entity;
+    PF_ARRAY PlayFabProfilesEntityPermissionStatement** statements;
+    PF_COLLECTION_COUNT uint32_t statementsCount;
+
+} PlayFabProfilesSetEntityProfilePolicyRequest;
+
+typedef struct PlayFabProfilesSetEntityProfilePolicyResponse
+{
+    PF_ARRAY PlayFabProfilesEntityPermissionStatement** permissions;
+    PF_COLLECTION_COUNT uint32_t permissionsCount;
+
+} PlayFabProfilesSetEntityProfilePolicyResponse;
+
+typedef struct PlayFabProfilesSetGlobalPolicyRequest
+{
+    PF_MAP struct PlayFabStringDictionaryEntry* customTags;
+    PF_COLLECTION_COUNT uint32_t customTagsCount;
+    PF_ARRAY PlayFabProfilesEntityPermissionStatement** permissions;
+    PF_COLLECTION_COUNT uint32_t permissionsCount;
+
+} PlayFabProfilesSetGlobalPolicyRequest;
+
+typedef struct PlayFabProfilesSetGlobalPolicyResponse
+{
+
+} PlayFabProfilesSetGlobalPolicyResponse;
+
```

Comment:
blank structs are odd.  i assume they won't appear in public headers?  if so that's fine

Reply:
ya, I agree this is a bit odd. Will probably need some special logic to remove these during autogen but that shouldn't be a problem

---
In reply to: [604528311](https://github.com/jplafonta/XPlatCSdk/pull/1#discussion_r604528311) [](ancestors = 604528311)